### PR TITLE
[ROCm] Change ROCm documentation

### PR DESCRIPTION
**Pull request by Ruturaj4**
* Fix Documentation to change ROCm JAX build/ installation instructions.

### CODE REVIEW

**Comment by Ruturaj4** [comment]
@superbobry @hawkinsp could please take a look? Thanks!

File: build/rocm/README.md [hunk 1]
```diff
@@ -5,34 +5,37 @@ This directory contains files and setup instructions to build and test JAX for R
...
-However, due to strict naming PyPI requirement we had to name our wheels slightly differently. This would then result in Jax/Jaxlib dependent not recognizing jaxlib-rocm wheels and would end up with multiple jaxlib installations and also runtime issues
-
-
-***
-### XLA for JAX ROCm
-We strive to push all ROCm related changes to the OpenXLA repository. However, at times some XLA changes for ROCm may not be upstreamed to XLA repo.Therefore, we have ROCm XLA branches that are associated with a Jaxlib release. These are available in ROCm fork of XLA here https://github.com/ROCmSoftwarePlatform/xla. See branches named as rocm-jaxlib-[jaxlib version]. For example, for jaxlib-v0.4.10, the branch is named rocm-jaxlib-v0.4.10. See path https://github.com/ROCmSoftwarePlatform/xla/tree/rocm-jaxlib-v0.4.10
-
 
+However, due to strict naming requirements by PyPI, we had to name our ROCm-specific wheels slightly differently. As a result, JAX/
+jaxlib dependencies may not recognize the jaxlib-rocm wheels, leading to multiple jaxlib installations and potential runtime issues.
```
Comment (superbobry) [inline build/rocm/README.md:41]: Are you sure this is indeed an issue? I think everything should work fine as long as the package your wheels ship is called `jaxlib`.

File: docs/developer.md [hunk 2]
```diff
@@ -195,28 +195,32 @@ sudo apt install miopen-hip hipfft-dev rocrand-dev hipsparse-dev hipsolver-dev \
...
-git clone https://github.com/ROCmSoftwarePlatform/xla.git
+git clone https://github.com/ROCm/xla.git
 ```
 
 and override the XLA repository with which JAX is built:
 
 ```
-python build/build.py --enable_rocm --rocm_path=/opt/rocm-5.7.0 \
-  --bazel_options=--override_repository=xla=/path/to/xla-rocm
+python3.10 ./build/build.py --use_clang=true --clang_path=/usr/lib/llvm-18/bin/clang-18 --enable_rocm --build_gpu_plugin --gpu_plugin_rocm_version=60 --bazel_options=--override_repository=xla=/rel/xla/ --rocm_path=/opt/rocm-6.2.3
```
Comment (superbobry) [inline docs/developer.md:219]: Is it important to use `python3.10` here?

File: build/rocm/README.md (hunk 1 above)
Comment (mrodden) [reply build/rocm/README.md:41]: Yeah I am pretty sure this is no longer an issue.

File: docs/developer.md [hunk 3]
```diff
@@ -195,28 +195,32 @@ sudo apt install miopen-hip hipfft-dev rocrand-dev hipsparse-dev hipsolver-dev \
...
 
+The recommended way to install these dependencies is by running our script, `jax/build/rocm/tools/get_rocm.py`, 
+and selecting the appropriate options.
+
 To build jaxlib with ROCM support, you can run the following build command,
 suitably adjusted for your paths and ROCM version.
 
 ```
-python build/build.py --enable_rocm --rocm_path=/opt/rocm-5.7.0
+python3.10 ./build/build.py --use_clang=true --clang_path=/usr/lib/llvm-18/bin/clang-18 --enable_rocm --build_gpu_plugin --gpu_plugin_rocm_version=60 --rocm_path=/opt/rocm-6.2.3
```
Comment (mrodden) [inline docs/developer.md:205]: Just leave as `python3`

The clang path is going to be system dependent unfortunately, but I guess we could leave it up to the user to change the example path here.

```suggestion
python3 ./build/build.py --use_clang=true --clang_path=/usr/lib/llvm-18/bin/clang-18 --enable_rocm --build_gpu_plugin --gpu_plugin_rocm_version=60 --rocm_path=/opt/rocm-6.2.3
```

File: docs/developer.md (hunk 2 above)
Comment (mrodden) [reply docs/developer.md:219]: Use `python3` please

```suggestion
python3 ./build/build.py --use_clang=true --clang_path=/usr/lib/llvm-18/bin/clang-18 --enable_rocm --build_gpu_plugin --gpu_plugin_rocm_version=60 --bazel_options=--override_repository=xla=/rel/xla/ --rocm_path=/opt/rocm-6.2.3
```

**Review by mrodden — CHANGES_REQUESTED** [review]

**Comment by Ruturaj4** [comment]
@superbobry and @mrodden thanks for the suggestions. Could you upstream please? Thanks!

File: build/rocm/README.md [hunk 4]
```diff
@@ -5,34 +5,34 @@ This directory contains files and setup instructions to build and test JAX for R
...
+```Bash
+docker run -it -d --network=host --device=/dev/kfd --device=/dev/dri --ipc=host --shm-size 64G --group-add video --cap-add=SYS_PTRACE --security-opt seccomp=unconfined -v ./:/jax --name rocm_jax jax-rocm:latest /bin/bash
 ```
 
 ***
 ### JAX ROCm Releases
-We strive to push all ROCm related changes to the OpenXLA repository. However, at times some JAX/JAXLIB changes for ROCm may not be present in upstream JAX repo.Therefore, we have ROCm Jax/Jaxlib branches that are associated with a Jaxlib release. These
-are available in ROCm fork of JAX https://github.com/ROCmSoftwarePlatform/jax. See branches named as rocm-jaxlib-[jaxlib-version]. For examples, for jaxlib-v0.4.10, the branch is named rocm-jaxlib-v0.4.10. See path https://github.com/ROCmSoftwarePlatform/jax/tree/rocm-jaxlib-v0.4.10
+We aim to push all ROCm-related changes to the OpenXLA repository. However, there may be times when certain JAX/JAXLIB updates for
+ROCm are not yet reflected in the upstream JAX repository. To address this, we maintain ROCm-specific JAX/JAXLIB branches tied to JAX
```
Comment (superbobry) [inline build/rocm/README.md:24]: Nit: we don't normally capitalize JAXLIB and instead use \`jaxlib\`.

Please update other references if you update this one.

**Review by superbobry — APPROVED** [review]

File: build/rocm/README.md (hunk 4 above)
Comment (Ruturaj4) [reply build/rocm/README.md]: Done @superbobry thanks!

**Pull request merged**